src/router/index.ts

```diff
@@ -20,12 +20,12 @@ const router = createRouter({
       component: () => import("@/views/SuccessView.vue")
     },
     {
-      path: "/:categoryName/:productId",
+      path: "/categories/:categoryName/:productId",
```

Comment:
Otherwise handling 404 would be a bit messy, so I decided to not overcomplicate. =/